torch/lib/c10d/ProcessGroupGloo.hpp

```diff
@@ -69,12 +69,16 @@ class ProcessGroupGloo : public ProcessGroup {
   // operations using the new AsyncWork base class. Over time we will port
   // all operations and perform needed cleanup.
   //
+  // FIXME: This probably should be called WorkGloo since the work is executed in sync mode
+  // by a background thread.
   class AsyncWork : public ProcessGroup::Work {
    public:
     AsyncWork(
+        std::vector<std::vector<at::Tensor>> outputTensors,
         const char* profilingTitle = nullptr,
         const c10::optional<std::vector<at::Tensor>>& inputTensors = c10::nullopt)
-        : ProcessGroup::Work(-1, OpType::UNKNOWN, profilingTitle, inputTensors) {
+        : ProcessGroup::Work(-1, OpType::UNKNOWN, profilingTitle, inputTensors),
+          outputTensors_(outputTensors) {
     }
 
```

Comment:
Shouldn't we also override the virtual destructor?

Reply:
if you define virtual destructor in a class it'll be always virtual in subclasses (and will be = default). However, it does not hurt to define one as well.